CodeEdit/Features/SourceControl/Views/SourceControlStashView.swift

```diff
@@ -0,0 +1,125 @@
+                        .contentShape(Rectangle())
+                        .frame(height: 48)
+                } header: {
+                    Text("Stash Changes")
+                    Group {
+                        if sourceControlManager.pullSheetIsPresented
+                            || sourceControlManager.switchToBranch != nil {
+                            Text("Your local repository has uncommitted changes that need to be stashed " +
+                                 "before you can continue. Enter a description for your changes.")
+                        } else {
+                            Text("Enter a description for your stashed changes so you can reference them later. " +
+                                 "Stashes will appear in the Source Control navigator for your repository.")
+                        }
+                    }
+                    .multilineTextAlignment(.leading)
+                    .lineLimit(nil)
+                }
+                if sourceControlManager.pullSheetIsPresented
+                    || sourceControlManager.switchToBranch != nil {
+                    Section {
+                        Toggle("Apply stash after operation", isOn: $applyStashAfterOperation)
+                    }
+                }
+            }
+            .formStyle(.grouped)
+            .scrollDisabled(true)
+            .scrollContentBackground(.hidden)
+            .onSubmit(submit)
+            HStack {
+                Spacer()
+                Button {
+                    message = ""
+                    dismiss()
+                } label: {
+                    Text("Cancel")
+                        .frame(minWidth: 56)
+                }
+                Button {
+                    submit()
+                } label: {
+                        Text(
+                            sourceControlManager.pullSheetIsPresented
+                            ? "Stash and Pull"
+                            : sourceControlManager.switchToBranch != nil
+                            ? "Stash and Switch"
+                            : "Stash"
+                        )
+                        .frame(minWidth: 56)
+                    }
+                .buttonStyle(.borderedProminent)
+            }
+            .padding(.horizontal, 20)
+            .padding(.bottom, 20)
+        }
+        .frame(width: 500)
+    }
+
+    func submit() {
+        Task {
+            do {
+                try await sourceControlManager.stashChanges(message: message)
+                message = ""
+
+                if sourceControlManager.pullSheetIsPresented
+                    || sourceControlManager.switchToBranch != nil {
+                    if sourceControlManager.pullSheetIsPresented {
+                        try await sourceControlManager.pull(
+                            remote: sourceControlManager.operationRemote?.name ?? nil,
+                            branch: sourceControlManager.operationBranch?.name ?? nil,
```

Comment:
The `?? nil` is redundant. Just use `sourceControlManager.operationBranch?.name`